### ISSUE TITLE
GGMAX Enhancement - NPC dialogue fades as you get further away

### ISSUE THREAD

**Issue by mike123yy** [TheGameCreators/GameGuruRepo#5635]
One thing I've noticed as of now is when you give the character dialogue that if you walk away from the character the audio still plays at maximum volume until the characters done speaking. So if the player activates the speech and decides to walk away halfway through it would be useful if the NPC dialogue would fade after a certain amount of distance or they would stop their dialogue. It would be really cool if that player chose to ignore the dialogue that it would play a second audio option after a certain distance away where you could make the NPC same something like "I didn't want to talk to you anyway". But as far as the fading audio at a distance it would also be helpful if you have two NPC's that are close proximity that their dialogue doesn't overlap.

**Comment by Blood-Moon-Interactive** [comment]
Are you using a mono or stereo sound file?

**Comment by mike123yy** [comment]
I opened them back up in audacity to check and they are in stereo. It wouldn't be hard to convert them to Mono but would that make a difference as far as it fading the sound as you get farther away?

**Comment by synchromesh62** [comment]
> I opened them back up in audacity to check and they are in stereo. It wouldn't be hard to convert them to Mono but would that make a difference as far as it fading the sound as you get farther away?

Probably not as they do not use 3D sound ,, There are 3D sound behaviours mainly for machines, radio's etc which you could blag using a trigger zone around a character but that would mean no lip-sync and an additional **Play Once** checkbox.  

![3D](https://github.com/user-attachments/assets/64bfda5d-6f40-4b60-afc7-6ab0208e72f7)

**Comment by mike123yy** [comment]
Yeah, I use all those for different things but without the lip-synching for characters using one of those wouldn't really cut it. If the character was wearing a mask or something that would do. If I was any good at Lua I would take a stab at mixing the 3-D sound script with the stand and speak script. It might be time I start learning though :-)

**Comment by Necrym59** [comment]
@mike123yy 
I thinks its handled in the master interpreter so lua wouldnt really help here i will mark it as an enhancement.
